src/App.tsx

```diff
@@ -17,18 +72,26 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setActivityFilter={setActivityFilter}
+                setQuery={setQuery}
+                query={todosQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!loadingDone
+              && (<Loader />)}
+              <TodoList
+                todos={filterTodos(todos, todosActivityFilter, todosQuery)}
+                todoFocusedOn={todoShown}
+                focusOnTodo={focusOnTodo}
+              />
```

Comment:
Use a ternary operator here, so you either render the Loader or TodoList depending on `loadingDone`

PS. I suggest renaming this piece of state to `isLoading`, more common convention

Reply:
Currently when the list is loading, you see loader and table at the same time